### DISCUSSION TITLE
HasOne doesnot work

### DISCUSSION THREAD

**Discussion by mirzaromi** [jenssegers/laravel-mongodb#2417] [General]
- Laravel-mongodb Version: 8.83.18
- PHP Version: 7.4.11
- Database Driver & Version: mongodb 4.4

### Steps to reproduce
1. 
2.
3.

### Expected behaviour
I want to store my data as a referece not embed but my relationship doesnt work

### Actual behaviour
store data as reference

<details><summary><b>Logs</b>:</summary>
Insert log.txt here (if necessary)
</details>
```
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
// use Illuminate\Database\Eloquent\Model;

use Jenssegers\Mongodb\Eloquent\Model;

use App\Models\Parents;

class Student extends Model
{
    use HasFactory;

    protected $connection = "mongodb";
    protected $collection = "student";
    protected $guarded = ["_id"];

    public function parents()
    {
        return $this->hasOne(Parents::class);
    }

}
```

**Comment by lbrutti** [comment]
hello, I don't know if that is you case, but i had a non-working hasOne (hybrid)relation that failed because I was not setting the $connection property in the related model...
